doc/cookbook/example_developer_workflow.md

```diff
@@ -1,49 +1,74 @@
+   Typically, Pachyderm users store the following artifacts in a
+   Git repository:
 
-Once your code is ready to commit to your git repo, here are the steps that can form the basis of a production workflow.
+   - A Dockerfile that you use to build local images.
+   - A `pipeline.json` specification file that you can use in a `Makefile` to
+   create local builds, as well as in the CI/CD workflows.
+   - The code that performs data transformations.
 
-### 1. Git commit hook
+   A [commit hook in Git](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
+   for your repository triggers the CI/CD process. It uses the
+   information in your pipeline specification for subsequent steps.
 
-A [commit hook in git](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) for your kicks off the
-Continous Integration/Continuous Deployment process.  It should  use the information present in a template for your Pachyderm pipelines for subsequent steps.
+2. Build an image
 
-### 2. Build image
+   Your CI process must automatically start the build of a Docker container
+   image based on your code and the Dockerfile.
 
-Your CI process should automatically kick off the build of an docker container image based on your code and the DOCKERFILE. That image will be used in the next step.
+3. Push the image tagged with commit ID to an image registry.
 
-### 3. Push to registry tagged with commit id
+   Your CI process pushes a Docker image created in Step 2 to your preferred
+   image registry. When you build your image by using  the `--built` flag with
+   the `pachctl update pipeline` command, Pachyderm tags the image with the
+   Git commit SHA.
 
-The docker image created in the prior step is then pushed to your preferred docker registry and tagged with the git commit SHA, shown as just "tag" in the figure above.
+4. Update the pipeline spec with the tagged image.
 
-### 4. 'update pipeline' using template with tagged image
+   In this step, your CI/CD infrastructure uses the `pipeline.json`
+   specification that you have checked in and fills in the Git commit
+   SHA for the version of the image that must be used in this pipeline.
+   Then, it runs the `pachctl update pipeline` command to push the
+   updated pipeline specification to Pachyderm.
 
-In this step, your CI/CD infrastructure would use the pipeline.json template that was checked in, and fill in the git commit SHA for the version of the image that should be used in this pipeline.  It will then use the ``pachctl update pipeline`` command to push it to pachyderm.
+5. Pull a tagged image from a registry
```

Comment:
This is not really a step. After CI calls update pipeline, this is jsut want pachyderm does. Probably should be combined with step 4 as just the "result" of that action.